Python/Tests/Analysis/AnalysisTest.cs

```diff
@@ -6804,16 +6805,18 @@ class Y:
                 new { Code="import abc", Index=10, Expected="abc", Base="" },
                 new { Code="import deg, abc as A", Index=12, Expected="abc", Base="" },
                 new { Code="from abc import A", Index=6, Expected="abc", Base="" },
-                new { Code="from .deg import A", Index=9, Expected="abc.deg", Base="abc" },
-                new { Code="from .hij import A", Index=9, Expected="abc.deg.hij", Base="abc.deg" },
-                new { Code="from ..hij import A", Index=10, Expected="abc.hij", Base="abc.deg" },
-                new { Code="from ..hij import A", Index=10, Expected="abc.deg.hij", Base="abc.deg.HIJ" },
+                new { Code="from .deg import A", Index=9, Expected="deg.A", Base="abc" },
+                new { Code="from .hij import A", Index=9, Expected="abc.hij.A", Base="abc.deg" },
+                new { Code="from ..hij import A", Index=10, Expected="hij.A", Base="abc.deg" },
+                new { Code="from ..hij import A", Index=10, Expected="abc.hij.A", Base="abc.deg.HIJ" },
+                new { Code="from .. import deg", Index=10, Expected="deg", Base="abc.deg" },
+                new { Code="from . import A", Index=10, Expected="abc.A", Base="abc.deg" },
```

Comment:
These tests are all incorrect now - you're dropping too many levels when processing the leading `.`.

I suggest creating these packages and testing everything to prove to yourself that Python treats `.` as "current package" in all cases. Use `print(__name__)` to see the order they get imported in.

Reply:
Ah, you're right. I was assuming (and testing) in the `__init__.py` files in the package, but those will always include `__init__` as the last part of the module name and so the logic still checks out.

Thanks for fixing my error here!